src/main/kotlin/com/openlattice/apps/services/AppService.kt

```diff
@@ -366,6 +369,23 @@ class AppService(
         apps.executeOnKey(appId, UpdateAppRolePermissionsProcessor(roleId, permissions))
     }
 
+    fun updateAppConfigRoleMapping(
+            appId: UUID,
+            organizationId: UUID,
+            roleMappings: Map<UUID, AclKey>
+    ) {
+        ensureAppRolesExist(getApp(appId), roleMappings.keys)
+        principalsService.ensurePrincipalsExist(roleMappings.values.toSet())
+
+        appConfigs.executeOnKey(AppConfigKey(appId, organizationId)) {
+            val setting = it.value
+            roleMappings.forEach { (roleId, roleAclKey) ->
```

Comment:
Make sure the map itself is always serializable. Had problems in Organization Service when folks were passing in a map that wasn't serializable.